src/Sarif/Core/Run.cs

```diff
@@ -17,6 +17,11 @@ public partial class Run
 
         private IDictionary<FileLocation, int> _fileToIndexMap;
 
+        public Uri GetExpandedUriBaseIdValue(string key, string currentValue = null)
+        {
+            throw new InvalidOperationException("Author this code along with tests");
+        }
+
         public int GetFileIndex(
```

Comment:
>GetFileIndex [](start = 19, length = 12)

this whole helper needs to be reconsidered in light of file parent index. and so i suspended unit testing.